### PR TITLE
Add support for merge keys

### PR DESCRIPTION
**Pull request by farcaller**
Adds support for merge keys and fixes https://github.com/dart-lang/yaml/issues/121.

(this is a copy of the closed #129)

### CODE REVIEW

**Comment by farcaller** [comment]
@mosuem PTAL? The CLA bot says that I have the signature on file since #129 was created.

**Comment by mosuem** [comment]
Could you copy the description and linked issue into this PR description, add a changelog entry and some tests for this new feature?

File: lib/src/loader.dart [hunk 1]
```diff
@@ -174,6 +174,18 @@ class Loader {
       children[key] = value;
       event = _parser.parse();
     }
+    if (children.containsKey('<<')) {
+      final mergeVal = children['<<']!;
+      if (mergeVal is! YamlMap) {
```
Comment (jonasfj) [inline lib/src/loader.dart:179]: Supposedly the draft says:
```yaml
---
- &CENTER { x: 1, y: 2 }
- &LEFT { x: 0, y: 2 }
- &BIG { r: 10 }
- &SMALL { r: 1 }

# All the following maps are equal:

- # Explicit keys
  x: 1
  y: 2
  r: 10
  label: center/big

- # Merge one map
  << : *CENTER
  r: 10
  label: center/big

- # Merge multiple maps
  << : [ *CENTER, *BIG ]
  label: center/big

- # Override
  << : [ *BIG, *LEFT, *SMALL ]
  x: 1
  label: center/big
```

Last two cases are not handled: https://yaml.org/type/merge.html

**Review by jonasfj — CHANGES_REQUESTED** [review]

**Comment by jonasfj** [comment]
I propose that this feature should not be implemented in `package:yaml`, see https://github.com/dart-lang/yaml/issues/121#issuecomment-2154574354

**Comment by mosuem** [comment]
Closing as the [dart-lang/yaml](https://github.com/dart-lang/yaml) repository is merged into the [dart-lang/tools](https://github.com/dart-lang/tools) monorepo. Please re-open this PR there!

**Pull request closed without merging**